assets/schema_input.json

```
@@ -41,7 +41,7 @@
             },
             "rundir": {
                 "type": "string",
-                "format": "file-path",
+                "format": "directory-path",
```

👏 That was fast! It works now!

---

While this change fixed the issue when running locally, it now give the following error when running against [samplesheet](https://raw.githubusercontent.com/KarNair/test-datasets/seqinspector/testdata/MiSeq/samplesheet.csv) hosted on github.

```
ERROR ~ ERROR: Validation of 'input' file failed!

 -- Check '.nextflow.log' file for details
The following errors have been detected:

* -- Entry 1 - rundir: Run directory must be a path (https://github.com/KarNair/test-datasets/raw/seqinspector/testdata/MiSeq/220422_M11111_0222_000000000-K9H97/)
 
```

---

Found this in the .nextflow.log
```
Mar-19 10:55:42.466 [main] DEBUG nextflow.validation.SchemaValidator - * -- Entry 1 - rundir: 'https://github.com/KarNair/test-datasets/raw/seqinspector/testdata/MiSeq/220422_M11111_0222_000000000-K9H97/' is not a directory, but a file (https://github.com/KarNair/test-datasets/raw/seqinspector/testdata/MiSeq/220422_M11111_0222_000000000-K9H97/)
```

---

I'll ask in #nf-validation

---

I think I remember this being an issue that if one wants to retrieve something from `http`, then it has to be a file. We may need to tarball this, and do stuff.

---

I remembered this was also why globs are not suitable for samplesheets as file paths because it doesn't work over `http`

---

So basically we got the response that this is not supported. Would it be ok to just specify this as a `path`, and then we can check in the workflow if this value is not a directory, then pass it through an untar process, and then join it together with the meta map to make the required tuple. Either way we're going to need some channel magic. Which processes are receiving the rundir?

---

It will be the instrument specific processes that receive this directory, so maybe it's fine for now. I think in nf-core/demultiplex that's exactly how they do it, i.e. they have an option to untar the directory (see https://github.com/nf-core/demultiplex/blob/aa4d93673a4e04b1daf7b4bd71269d6054275534/workflows/demultiplex.nf#L101)

---

Yeah I agree with your idea @mahesh-panchal. 👍

---

```suggestion
                "format": "path",
```

This fixes the problem of running locally and against github as well. Not the ideal solution but this works. This will definitely make the CI gods happy

---

I'm fixing that now using `oneOf`